yw_website/yw_website/apps/website/utils.py

```diff
@@ -89,3 +90,24 @@ def get_direct_descendants(program_block_id, program_blocks):
         new_child.direct_descendants = get_direct_descendants(new_child.id, program_blocks)
         descendants.append(new_child)
     return descendants
+
+def get_ports(parents):
```

Comment:
since we want to use this function in the `get_direct_descendants` function, it would probably be more useable if
we just pass in a single block object and removed the for loop

the in get_direct_descendants you cold call in the for loop once and we'd be good to go

Reply:
I'll do that